worker/src/RTC/DtlsTransport.cpp

```diff
@@ -988,17 +988,22 @@ namespace RTC
 		uv_timeval_t dtlsTimeout{ 0, 0 };
 		uint64_t timeoutMs;
 
-		// NOTE: If ret == 0 then ignore the value in dtlsTimeout.
 		// NOTE: No DTLSv_1_2_get_timeout() or DTLS_get_timeout() in OpenSSL 1.1.0-dev.
 		ret = DTLSv1_get_timeout(this->ssl, static_cast<void*>(&dtlsTimeout)); // NOLINT
 
 		if (ret == 0)
+		{
+			OnTimer(this->timer);
+
 			return true;
+		}
 
 		timeoutMs = (dtlsTimeout.tv_sec * static_cast<uint64_t>(1000)) + (dtlsTimeout.tv_usec / 1000);
 
 		if (timeoutMs == 0)
 		{
+			OnTimer(this->timer);
```

Comment:
Here we are calling `OnTimer()` explicitly without considering whether the timer is already armed.

`OnTimer()` may call `Reset()` in which case `this->ssl` get's invalidated. Then the legitimately armed timer expires and `OnTimer()` get's called again, calling `DTLSv1_handle_timeout(this->ssl);`  with the invalidated `this->ssl`

Reply:
well noted, yes.